ext/nokogiri/extconf.rb

```diff
@@ -928,7 +928,10 @@ def configure
     end
 
     if darwin? && !cross_build_p
-      recipe.configure_options += ["RANLIB=/usr/bin/ranlib", "AR=/usr/bin/ar"]
+      recipe.configure_options += [
+        "RANLIB=#{ENV.fetch("RANLIB", "/usr/bin/ranlib")}",
```

Comment:
For libxml2's `configure` script, if an environment variable like `RANLIB` is set, then it doesn't need to be passed explicitly to the configure command. Can you restructure this to something like the following

```ruby
recipe.configure_options << "RANLIB=/usr/bin/ranlib" unless ENV.key?("RANLIB")
recipe.configure_options << "AR=/usr/bin/ar" unless ENV.key?("AR")
```